sklearn/metrics/tests/test_classification.py

```diff
@@ -818,7 +819,9 @@ def test_zero_division_nan_no_warning(metric, y_true, y_pred, zero_division):
     """
     with warnings.catch_warnings():
         warnings.simplefilter("error")
-        result = metric(y_true, y_pred, zero_division=zero_division)
+        # Ignore warnings related to wrong shape of confusion_matrix:
+        with ignore_warnings(category=UserWarning):
```

Comment:
I'm not so happy with this solution, because this test should either not bother about the `UndefinedMetricWarning` at all (because the other test below already does), of if it does, then we should ignore only this very specific `UserWarnings` coming from `confusion_matrix`.